doc/rst/developer/bestPractices/TestSystem.rst

```diff
@@ -466,28 +637,178 @@ include a future for the issue tracked on the `GitHub issues page`_.
+**Test Settings**
+-------------------------------------------------------------------------------
+foo.compopts        line separated compiler flag configurations.  See
+                    `Compile-time Arguments`_ for more information
+COMPOPTS            directory-wide compiler flags
+foo.execopts        line separated runtime flag configurations.  See
+                    `Execution-time Arguments`_ for more information
+EXECOPTS            directory-wide runtime flags
+foo.execenv         line separated list of environment variables settings.  See
+                    `Environment Variables`_ for more information
+EXECENV             directory-wide environment variables
+foo.numlocales      number of locales to use in multi-locale run
+NUMLOCALES          directory-wide number of locales to use in multi-locale run
+..
+-------------------------------------------------------------------------------
+**Helper files**
+-------------------------------------------------------------------------------
+foo.catfiles        line separated list of files to include when validating the
+                    expected output
+CATFILES            directory-wide list of files to compare with output
+foo.prediff         script that is run on the test output, before taking the
+                    diff between the output and .good file
+PREDIFF             directory-wide script that is run over test output
+foo.precomp         script that is run prior to compilation of the test program
+PRECOMP             directory-wide script that is run prior to compilation
+foo.preexec         script that is run prior to execution of the test program
+PREEXEC             directory-wide script that is run prior to execution
+..
+-------------------------------------------------------------------------------
+**Testing System Settings**
+-------------------------------------------------------------------------------
+foo.cleanfiles      line separated list of files to remove before next test run
+CLEANFILES          directory-wide list of files to remove before test runs
+foo.noexec          empty file. Indicates .chpl file should only be compiled,
+                    not executed.  See `Controlling How It Runs`_ for more
+                    information.
+foo.notest          empty file. Indicates the file should not be run explicitly
+                    See `Controlling How It Runs`_ for more information.
+NOTEST              empty file. Indicates the directory should not be run
+foo.skipif          line separated list of conditions under which the test
+                    should not be run, or a script to compute the same.  See
+                    `Limiting Where the Test Runs`_ for more information
+SKIPIF              same as above, but applied to the entire directory
+foo.suppressif      line separated list of conditions under which the test is
+                    expected to fail, or a script to compute the same
```

Comment:
It's pretty subjective, and we're not that consistent ourselves. Generally a supressif is for something we expect/want to fail forever in one config (pretty rare), or something that was passing and is now failing for some reason (usually outside our control), but we expect/hope it to work in the future. 

For some examples see https://github.com/chapel-lang/chapel/pull/8258 or https://github.com/chapel-lang/chapel/pulls?q=is%3Apr+suppressif+is%3Aclosed+author%3Aronawho

Reply:
> something that was passing and is now failing for some reason (usually outside our control), but we expect/hope it to work in the future

One example of this is a failure due to a bug in a target compiler (gcc, icc, pgi), or a bug in a package

Reply:
Because the use case of `suppressif` is so variable (and not necessarily well defined), Elliot and I thought it best not to give `suppressif` its own section.  @cassella - would you prefer we defined it more thoroughly here?

Reply:
Maybe just enough that someone reading the doc to write a test would know which to use.  Or at least, would get the impression to use a ``.skipif`` unless a reviewer says otherwise.